api_tester/lib/purchases_flutter_api_test.dart

```diff
@@ -0,0 +1,420 @@
+// ignore_for_file: deprecated_member_use
+class _PurchasesFlutterApiTest {
+
+  void _checkSetup() {
+    String apiKey = "fakeApiKey";
+    String? userId = "fakeUserId";
+    bool observerMode = false;
+    String? userDefaultsSuiteName = "fakeSuiteName";
+    bool useAmazon = false;
+    Future<void> callback = Purchases.setup(
+      apiKey,
+      appUserId: userId,
+      observerMode: observerMode,
+      userDefaultsSuiteName: userDefaultsSuiteName,
+      useAmazon: useAmazon
+    );
+  }
+
+  void _checkConfigure() {
+    PurchasesConfiguration configuration = PurchasesConfiguration("fakeApiKey");
+    Future<void> callback = Purchases.configure(configuration);
+  }
+
+  void _checkSetFinishTransactions() {
+    bool finishTransactions = false;
+    Future<void> callback = Purchases.setFinishTransactions(finishTransactions);
+  }
+
+  void _checkSetAllowSharingStoreAccount() {
+    bool allowSharing = false;
+    Future<void> callback = Purchases.setAllowSharingStoreAccount(allowSharing);
+  }
+
+  void _checkAddCustomerInfoUpdateListener() {
+    void customerInfolistener(CustomerInfo customerInfo) {}
+    Purchases.addCustomerInfoUpdateListener(customerInfolistener);
+  }
+
+  void _checkRemoveCustomerInfoUpdateListener() {
+    void customerInfolistener(CustomerInfo customerInfo) {}
+    Purchases.removeCustomerInfoUpdateListener(customerInfolistener);
+  }
+
+  void _checkAddReadyForPromotedProductPurchaseListener() {
+    void listener(
+        String productIdentifier,
+        Future<PromotedPurchaseResult> Function() startPurchase
+        ) {}
+    Purchases.addReadyForPromotedProductPurchaseListener(listener);
+  }
+
+  void _checkRemoveReadyForPromotedProductPurchaseListener() {
+    void listener(
+        String productIdentifier,
+        Future<PromotedPurchaseResult> Function() startPurchase
+        ) {}
+    Purchases.removeReadyForPromotedProductPurchaseListener(listener);
+  }
+
+  void _checkGetOfferings() async {
+    Future<Offerings> offeringsFuture = Purchases.getOfferings();
+    Offerings offerings = await Purchases.getOfferings();
+  }
+
+  void _checkGetProducts() async {
+    List<String> productIdentifiers = List.empty();
+    PurchaseType purchaseType = PurchaseType.subs;
+    Future<List<StoreProduct>> productsFuture = Purchases.getProducts(
+        productIdentifiers,
+        type: purchaseType
+    );
+    productsFuture = Purchases.getProducts(
+        productIdentifiers
+    );
+    List<StoreProduct> products = await Purchases.getProducts(
+        productIdentifiers
+    );
+  }
+
+  void _checkPurchaseProduct() async {
+    String productIdentifier = "fakeProductId";
+    UpgradeInfo? upgradeInfo;
+    PurchaseType purchaseType = PurchaseType.subs;
+    Future<CustomerInfo> customerInfoFuture = Purchases.purchaseProduct(
+      productIdentifier,
+      upgradeInfo: upgradeInfo,
+      type: purchaseType
+    );
+    customerInfoFuture = Purchases.purchaseProduct(
```

Comment:
When there are optional parameters, I'm calling the method multiple times to make sure the different options are valid